src/nile/core/types/transactions.py

```diff
@@ -60,14 +60,18 @@ class Transaction(ABC):
 
     def __post_init__(self):
         """Populate pending fields."""
+        # Normalize max_fee if None
+        if self.max_fee is None:
+            self.max_fee = 0
+
```

Comment:
```suggestion
```

With `max_fee: int = 0` we don't need to normalize the max_fee (will be set to 0 even when None is passed)